pkg/rbac/parser.go

```diff
@@ -200,6 +222,53 @@ func GenerateRoles(ctx *genall.GenerationContext, roleName string) ([]interface{
 			ruleMap[key].addVerbs(rule.Verbs)
 		}
 
+		// deduplicate resources
+		// 1. create map based on key without resources
+		ruleMapWithoutResources := make(map[string][]*Rule)
+		for _, rule := range ruleMap {
+			// unset Resources on the key
+			key := rule.KeyWithoutResources()
+			if _, ok := ruleMapWithoutResources[key]; !ok {
+				rules := make([]*Rule, 0)
+				ruleMapWithoutResources[key] = rules
+			}
+			ruleMapWithoutResources[key] = append(ruleMapWithoutResources[key], rule)
+		}
+		// 2. merge to ruleMap
+		ruleMap = make(map[ruleKey]*Rule)
+		for _, rules := range ruleMapWithoutResources {
+			rule := rules[0]
+			for _, mergeRule := range rules[1:] {
+				rule.Resources = append(rule.Resources, mergeRule.Resources...)
```

Comment:
Isn't it a problem that we are losing Namespace of rules[1:]?

Reply:
NormalizeRules gets executed below for a single namespace. So all rules passed to this function act on the same namespace.